### PR TITLE
enhance mii-1.1.0 recipe

### PR DESCRIPTION
**Pull request by JLague**
- use SYSTEM toolchain
- check the shell when loading module
- move index file location

### CODE REVIEW

File: easybuild/easyconfigs/m/mii/mii-1.1.0.eb [hunk 1]
```diff
@@ -34,13 +34,19 @@ sanity_check_paths = {
...
 local hook_name = "command_not_found_handle"
 local hook_def = "source " .. pathJoin(root, "share/mii/init/common") .. " $@"
 
--- Define hooks for both bash and zsh
+-- Correct the hook name if shell is zsh
+if (myShellName() == "zsh") then
+    hook_name = hook_name .. 'r'
+end
+
+-- Define command not found hook
```
Comment (mboisson) [inline easybuild/easyconfigs/m/mii/mii-1.1.0.eb]: Should we give an error if the shell is not zsh or bash ?

Comment (mboisson) [reply easybuild/easyconfigs/m/mii/mii-1.1.0.eb]: Something like
```
if (myShellName() == "zsh" or myShellName() == "bash") then
    set_shell_function(hook_name, hook_def, "")
else
    if (mode() == "load") then
        LmodError("This module can not be used with shells other than bash or zsh")
    end
end

Comment (JLague) [reply easybuild/easyconfigs/m/mii/mii-1.1.0.eb]: Mii can work with other shells, but the shell integration only supports `bash` and `zsh`. We could register the command not found hook only if the shell is `bash` or `zsh`

Comment (JLague) [reply easybuild/easyconfigs/m/mii/mii-1.1.0.eb]: For example, `mii search blastn` can be used even with `tcsh` to find which modules provide the `blastn` command.

Comment (mboisson) [reply easybuild/easyconfigs/m/mii/mii-1.1.0.eb]: good point.

**Pull request merged**